typescript/sdk/src/core/read.ts

```diff
@@ -0,0 +1,54 @@
+import { providers } from 'ethers';
+
+import { Mailbox__factory } from '@hyperlane-xyz/core';
+import { Address, objMap, promiseObjAll } from '@hyperlane-xyz/utils';
+
+import { EvmHookReader } from '../hook/read.js';
+import { EvmIsmReader } from '../ism/read.js';
+import { MultiProvider } from '../providers/MultiProvider.js';
+import { ChainName } from '../types.js';
+
+import { CoreConfig } from './types.js';
+
+export class EvmCoreReader {
+  provider: providers.Provider;
+  evmHookReader: EvmHookReader;
+  evmIsmReader: EvmIsmReader;
+  constructor(
+    protected readonly multiProvider: MultiProvider,
+    chain: ChainName,
+    protected readonly concurrency: number = 20,
```

Comment:
Two minor things:
1) Let's make these consistent. I don't think there's a reason they can't be `public`
2) Let's make a DEFAULT_CONTRACT_READ_CONCURRENCY const somewhere if there isn't one yet. I noticed @paulbalaji had 20 hard-coded in a few places too the other day

Reply:
const for default reading concurrency would be great - after merging I might tweak the logic here slightly with the move of concurrency into chainmetadata

Reply:
Fixed.

Reply:
I've created a const/crud.ts for DEFAULT_CONTRACT_READ_CONCURRENCY. Feel free to update/remove if needed